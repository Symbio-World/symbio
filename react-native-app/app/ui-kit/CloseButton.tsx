import * as React from 'react'

import { RoundedButton } from './RoundedButton'
import { ButtonStyle } from './Button'
import { Icons } from './Icon'

type CloseButtonProps = {
  onClose?: () => void
}

export const CloseButton: React.FC<CloseButtonProps> = ({ onClose }) => (
  <RoundedButton
    onPress={onClose}
    outerTitle="close"
    variant={ButtonStyle.OUTLINE}
    icon={Icons.CROSS}
  />
)
