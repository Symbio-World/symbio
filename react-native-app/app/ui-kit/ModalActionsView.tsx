import * as React from 'react'
import { t } from 'react-native-tailwindcss'
import { View } from 'react-native'

type ModalActionsViewProps = {
  children: React.ReactNode
}

export const ModalActionsView: React.FC<ModalActionsViewProps> = ({
  children,
}) => (
  <View
    style={[
      t.flexRow,
      t.justifyEvenly,
      t.absolute,
      t.bottom0,
      t.wFull,
      t.itemsCenter,
    ]}
  >
    {children}
  </View>
)
