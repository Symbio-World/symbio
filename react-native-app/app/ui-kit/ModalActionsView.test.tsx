import * as React from 'react'
import { render } from 'react-native-testing-library'
import { ModalActionsView } from './ModalActionsView'
import { CloseButton } from './CloseButton'

describe('ModalActionsView', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <ModalActionsView>
        <CloseButton />
      </ModalActionsView>,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
