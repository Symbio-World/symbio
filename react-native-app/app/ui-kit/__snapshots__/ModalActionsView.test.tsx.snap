// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalActionsView renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "justifyContent": "space-evenly",
      },
      Object {
        "position": "absolute",
      },
      Object {
        "bottom": 0,
      },
      Object {
        "width": "100%",
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "padding": 8,
          },
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "borderRadius": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "borderWidth": 2,
            },
            Object {
              "borderColor": "#B35340",
            },
            Object {
              "alignItems": "center",
            },
            Object {
              "justifyContent": "center",
            },
            Object {
              "width": 48,
            },
            Object {
              "height": 48,
            },
            Object {
              "borderRadius": 9999,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 20,
              },
              Object {
                "height": 20,
              },
            ]
          }
        >
          <SvgMock
            height="100%"
            width="100%"
          />
        </View>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "#B35340",
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          close
        </Text>
      </View>
    </View>
  </View>
</View>
`;
